pkg/operators/spoke/controller.go

```diff
@@ -2,28 +2,395 @@ package spoke
 
 import (
 	"context"
+	"fmt"
+	"path/filepath"
 
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
 	"k8s.io/client-go/kubernetes"
+	"k8s.io/klog"
 
+	"github.com/openshift/api"
 	"github.com/openshift/library-go/pkg/controller/factory"
 	"github.com/openshift/library-go/pkg/operator/events"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+
+	"github.com/openshift/library-go/pkg/assets"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceapply"
+	operatorhelpers "github.com/openshift/library-go/pkg/operator/v1helpers"
+
+	nucleusv1client "github.com/open-cluster-management/api/client/nucleus/clientset/versioned/typed/nucleus/v1"
+	nucleusinformer "github.com/open-cluster-management/api/client/nucleus/informers/externalversions/nucleus/v1"
+	nucleuslister "github.com/open-cluster-management/api/client/nucleus/listers/nucleus/v1"
+	nucleusapiv1 "github.com/open-cluster-management/api/nucleus/v1"
+	"github.com/open-cluster-management/nucleus/pkg/helpers"
+	"github.com/open-cluster-management/nucleus/pkg/operators/spoke/bindata"
 )
 
-type nucleusAgentController struct {
-	kubeClient kubernetes.Interface
+const (
+	nucleusSpokeFinalizer        = "nucleus.open-cluster-management.io/spoke-core-cleanup"
+	bootstrapHubKubeConfigSecret = "bootstrap-hub-kubeconfig"
+	hubKubeConfigSecret          = "hub-kubeconfig"
+	nucluesSpokeCoreNamespace    = "open-cluster-management"
+	spokeCoreApplied             = "Applied"
+)
+
+var (
+	genericScheme = runtime.NewScheme()
+	genericCodecs = serializer.NewCodecFactory(genericScheme)
+	genericCodec  = genericCodecs.UniversalDeserializer()
+)
+
+type nucleusSpokeController struct {
+	nucleusClient          nucleusv1client.SpokeCoreInterface
+	nucleusLister          nucleuslister.SpokeCoreLister
+	kubeClient             kubernetes.Interface
+	registrationGeneration int64
+	workGeneration         int64
+}
+
+func init() {
+	utilruntime.Must(api.InstallKube(genericScheme))
```

Comment:
Do we need to install the schemes for all of these APIs here?

Reply:
(I wouldn't want to take a dep on openshift API just for this method, especially if we only need it to install schemes)